cmake/modules/FindASS.cmake

```
@@ -10,59 +10,195 @@
-    set_target_properties(PkgConfig::PC_ASS PROPERTIES INTERFACE_LINK_OPTIONS "")
-
-    # First item is the full path of the library file found
-    # pkg_check_modules does not populate a variable of the found library explicitly
-    list(GET PC_ASS_LINK_LIBRARIES 0 ASS_LIBRARY)
-    set(ASS_INCLUDE_DIR ${PC_ASS_INCLUDEDIR})
-    set(ASS_VERSION ${PC_ASS_VERSION})
-  elseif(WIN32 OR WINDOWS_STORE)
-    find_package(libass CONFIG QUIET REQUIRED
-                        HINTS ${DEPENDS_PATH}/lib/cmake
-                        ${${CORE_PLATFORM_NAME_LC}_SEARCH_CONFIG})
+  macro(buildlibASS)
+
+    find_package(FreeType REQUIRED QUIET)
+    find_package(HarfBuzz REQUIRED QUIET)
+    find_package(FriBidi REQUIRED QUIET)
+    find_package(Iconv REQUIRED QUIET)
+
+    # Posix platforms (except Apple) use fontconfig
+    if(NOT (WIN32 OR WINDOWS_STORE) AND
+       NOT (CMAKE_SYSTEM_NAME STREQUAL Darwin))
+      find_package(Fontconfig REQUIRED QUIET)
+    endif()
+
+    set(ASS_VERSION ${${MODULE}_VER})
+
+    if(WIN32 OR WINDOWS_STORE)
+      set(patches "${CMAKE_SOURCE_DIR}/tools/depends/target/libass/01-win-CMakeLists.patch"
+                  "${CMAKE_SOURCE_DIR}/tools/depends/target/libass/02-win-dwrite-fontload.patch"
+                  "${CMAKE_SOURCE_DIR}/tools/depends/target/libass/03-win-dwrite-prevent-ass_set_fonts_dir.patch"
+                  "${CMAKE_SOURCE_DIR}/tools/depends/target/libass/04-win-nullcheck-shared_hdc.patch")
+
+      generate_patchcommand("${patches}")
+
+      set(LIBASS_DEBUG_POSTFIX d)
+    endif()
 
-    # we only do this because we use find_package_handle_standard_args for config time output
-    # and it isnt capable of handling TARGETS, so we have to extract the info
-    get_target_property(_ASS_CONFIGURATIONS libass::libass IMPORTED_CONFIGURATIONS)
-    foreach(_ass_config IN LISTS _ASS_CONFIGURATIONS)
-      # Some non standard config (eg None on Debian)
-      # Just set to RELEASE var so select_library_configurations can continue to work its magic
-      string(TOUPPER ${_ass_config} _ass_config_UPPER)
-      if((NOT ${_ass_config_UPPER} STREQUAL "RELEASE") AND
-         (NOT ${_ass_config_UPPER} STREQUAL "DEBUG"))
-        get_target_property(ASS_LIBRARY_RELEASE libass::libass IMPORTED_LOCATION_${_ass_config_UPPER})
-      else()
-        get_target_property(ASS_LIBRARY_${_ass_config_UPPER} libass::libass IMPORTED_LOCATION_${_ass_config_UPPER})
+    if(WIN32 OR WINDOWS_STORE)
+      set(CMAKE_ARGS -DDUMMY_DEFINE=ON
+                     ${LIBASS_ADDITIONAL_ARGS})
+    else()
+      find_program(AUTORECONF autoreconf REQUIRED)
+      if (CMAKE_HOST_SYSTEM_NAME MATCHES "(Free|Net|Open)BSD")
+        find_program(MAKE_EXECUTABLE gmake)
       endif()
-    endforeach()
+      find_program(MAKE_EXECUTABLE make REQUIRED)
+
+      set(CONFIGURE_COMMAND ${AUTORECONF} -vif
+                    COMMAND ./configure
+                              --prefix=${DEPENDS_PATH}
+                              --disable-shared)
```

Needed to disable `libunibreak`, otherwise linking failed for me on Arch with `libunibreak` installed on the system.
```suggestion
                    COMMAND ./configure
                              --prefix=${DEPENDS_PATH}
                              --disable-libunibreak
                              --disable-shared)
```